### PR TITLE
ci: add disk check to script

### PR DESCRIPTION
**Pull request by nathan-weinberg**
the github action was cleaning up the ilab artifacts
even if we preserved the E2E test dir

add disk checks into the script itself for more
granular checking + benefiting the libraries

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Functional tests have been added, if necessary.
- [ ] E2E Workflow tests have been added, if necessary.

### CODE REVIEW

**Comment by nathan-weinberg** [comment]
This did not seem to work based on the results from https://github.com/instructlab/instructlab/actions/runs/12148323866/job/33876674159?pr=2738 (no change in disk space)

**Comment by nathan-weinberg** [comment]
Changed the approach here, details in the commit desc

File: .github/workflows/e2e-nvidia-l4-x1.yml [hunk 1]
```diff
@@ -141,10 +141,6 @@ jobs:
           . venv/bin/activate
           ./scripts/e2e-ci.sh -m
 
-      - name: Check disk after tests
```
Comment (danmcp) [inline .github/workflows/e2e-nvidia-l4-x1.yml:144]: Would it be ok to leave these?  They might come in handy later.

**Review by bbrowning — APPROVED** [review]
Ahh, good catch and thanks for ensuring this works as expected!

File: .github/workflows/e2e-nvidia-l4-x1.yml (hunk 1 above)
Comment (bbrowning) [reply .github/workflows/e2e-nvidia-l4-x1.yml:144]: Just for clarity, are you asking for both to stay? The new one added by Nathan that runs during the CI script, and this one that runs after at least some of the CI-generated artifacts have been cleaned up?

Comment (nathan-weinberg) [reply .github/workflows/e2e-nvidia-l4-x1.yml:144]: Yes we can keep it for sure! Will update the patchset when I get in 👍

Comment (bbrowning) [reply .github/workflows/e2e-nvidia-l4-x1.yml:144]: Sounds reasonable - thanks for the explanation @danmcp :smile:

File: .github/workflows/e2e-nvidia-t4-x1.yml [hunk 2]
```diff
@@ -140,7 +140,7 @@ jobs:
       - name: Check disk after tests
         run: |
           df -h
-
```
Comment (danmcp) [inline .github/workflows/e2e-nvidia-t4-x1.yml:143]: nit: looks like a remnant was left around

**Review by danmcp — APPROVED** [review]

**Pull request merged**